packages/babel-cli/src/babel/dir.js

```diff
@@ -172,16 +186,29 @@ export default async function({
         },
       });
 
+      // This, alongside with debounce, allows us to only log
+      // when we are sure that all the files have been compiled.
+      let processing = 0;
+
       ["add", "change"].forEach(function(type: string): void {
-        watcher.on(type, function(filename: string): void {
-          handleFile(
-            filename,
-            filename === filenameOrDir
-              ? path.dirname(filenameOrDir)
-              : filenameOrDir,
-          ).catch(err => {
+        watcher.on(type, async function(filename: string) {
+          processing++;
+
+          try {
+            await handleFile(
+              filename,
+              filename === filenameOrDir
+                ? path.dirname(filenameOrDir)
+                : filenameOrDir,
+            );
+
+            compiledFiles++;
```

Comment:
Will `compiledFiles` be incremented twice if a single file is modified twice?

Reply:
With the current implementation, if it is modified twice in a very short period of time, yes.

Do you think that the memory/complexity of using a `Set` with the file paths instead of the simple counter? (This looks like a rhetorical question, but it's not :sweat_smile:)

Reply:
On devtools like Babel I prefer correctness over performance unless it becomes our bottleneck.

On the other hand, thinking of the DX, does `Successfully compiled # files with babel` really deliver value? I mean as a developer I would not care how many files are compiled, especially when in watch mode. In Babel repo we are printing the compiled path, execution timestamp in watch mode. If one is changing a single file using babel-cli, it will keep printing `successfully compiled 1 file with Babel` so babel-cli can not assure users that the file _has_ been compiled after they saved.

What about printing things like "successfully compiled after 346ms", not ideal but at least they are more likely to differ across different runs.

I am okay to leave it as-is and address this in a separate DX PR.

Reply:
Yeah I'm up for adjusting the message too, in a follow-up.

Reply:
Ok :+1:

Reply:
It now logs
```
Successfully compiled 3 files with Babel (123ms).
```